src/main/java/bio/terra/service/dataset/flight/LockDatasetStep.java

```diff
@@ -1,36 +1,55 @@
 package bio.terra.service.dataset.flight;
 
 import bio.terra.common.exception.RetryQueryException;
-import bio.terra.service.dataset.DatasetDao;
+import bio.terra.service.dataset.DatasetService;
 import bio.terra.service.dataset.exception.DatasetLockException;
 import bio.terra.service.dataset.exception.DatasetNotFoundException;
+import bio.terra.service.dataset.flight.ingest.OptionalStep;
 import bio.terra.stairway.FlightContext;
-import bio.terra.stairway.Step;
 import bio.terra.stairway.StepResult;
 import bio.terra.stairway.StepStatus;
 import java.util.UUID;
+import java.util.function.Predicate;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
-public class LockDatasetStep implements Step {
+public class LockDatasetStep extends OptionalStep {
 
   private static Logger logger = LoggerFactory.getLogger(LockDatasetStep.class);
 
-  private final DatasetDao datasetDao;
+  private final DatasetService datasetService;
   private final UUID datasetId;
   private final boolean sharedLock; // default to false
   private final boolean suppressNotFoundException; // default to false
 
-  public LockDatasetStep(DatasetDao datasetDao, UUID datasetId, boolean sharedLock) {
-    this(datasetDao, datasetId, sharedLock, false);
+  public LockDatasetStep(DatasetService datasetService, UUID datasetId, boolean sharedLock) {
+    this(datasetService, datasetId, sharedLock, false, OptionalStep::alwaysDo);
```

Comment:
Seems like chaining to the constructor without `doCondition` would work here
```suggestion
    this(datasetService, datasetId, sharedLock, false);
```